prometheus/prometheus.yaml

```diff
@@ -0,0 +1,215 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: prometheus
+  namespace: {{ default_ns.name }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: prometheus
+  namespace: {{ default_ns.name }}
+rules:
+ - apiGroups: [""]
+   resources:
+    - nodes
+    - nodes/proxy
+    - services
+    - endpoints
+    - pods
+   verbs: ["get", "list", "watch"]
+ - apiGroups:
+    - extensions
+   resources:
+    - ingresses
+   verbs: ["get", "list", "watch"]
+ - nonResourceURLs: ["/metrics"]
+   verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus
+subjects:
+ - kind: ServiceAccount
+   name: prometheus
+   namespace: {{ default_ns.name }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: etc-prometheus
+  namespace: {{ default_ns.name }}
+data:
+  prometheus.yml: |
+    global:
+      scrape_interval: 15s
+    scrape_configs:
+     - job_name: "kubernetes-kubelet"
+       scheme: https
+       tls_config:
+         ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+         insecure_skip_verify: true
```

Comment:
What does this line mean?

Reply:
it tells prometheus to skip the verification of the target's cert, without which prometheus rejects the scrape with `cannot validate certificate for <POD_IP> because it doesn't contain any IP SANs`. I haven't looked much into what it would take to remedy that yet

Reply:
I'm not read up on how our DNS works outside of kubernetes services and what's baked into our certs. This directly scrapes pods, not services, so I don't see how we can verify the name

Reply:
it is possible to scrape services as well / instead of pods

Reply:
Ah, this is because k8s certs don't have the pod IPs in them, OK, this seems fine.